### PR TITLE
java backend

### PR DESCRIPTION
**Pull request by arturoLucazzz**
falta integrar con next, al parecer hay un pequeño conflicto, podría verse en otra branch

### CODE REVIEW

File: java-backend/src/main/java/com/custom/todo/list/app/controller/TaskHttp.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Validated
+@RestController
+@RequestMapping(path = "/v1.0/api/todo")
```
Comment (HYAR7E) [inline java-backend/src/main/java/com/custom/todo/list/app/controller/TaskHttp.java:25]: Maybe we can remove the versioning here so no changes are required in the frontend?
```suggestion
@RequestMapping(path = "/api/todo")
```

File: frontend/src/utils/config.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export const API_HOST = process.env.NEXT_PUBLIC_API_HOST;
 export const todoAPIEndpoints = {
-  list: () => `${API_HOST}/api/todo`,
-  create: () => `${API_HOST}/api/todo`,
-  update: (id: number) => `${API_HOST}/api/todo/${id}/`,
-  delete: (id: number) => `${API_HOST}/api/todo/${id}/`,
+  list: () => `${API_HOST}/v1.0/api/todo`,
+  create: () => `${API_HOST}/v1.0/api/todo`,
+  update: (id: number) => `${API_HOST}/v1.0/api/todo/${id}/`,
+  delete: (id: number) => `${API_HOST}/v1.0/api/todo/${id}/`,
```
Comment (HYAR7E) [inline frontend/src/utils/config.ts:6]: Frontend should be API independant so any backend implementation shouldn't require changes in the frontend

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,9 @@
 up-node:
 	docker-compose -f docker-compose.yml -f docker-compose.node.yml up --build
 
+up-java:
+	mvn -f java-backend/ package
```
Comment (HYAR7E) [inline Makefile:5]: I'm not able to execute this on my local machine, maybe we can include this into the container?
image

File: java-backend/src/main/java/com/custom/todo/list/app/controller/TaskHttp.java (hunk 1 above)
Comment (arturoLucazzz) [reply java-backend/src/main/java/com/custom/todo/list/app/controller/TaskHttp.java:25]: i added it as a good practice, to give a good starting point to guyss, what do y think?

File: Makefile (hunk 3 above)
Comment (arturoLucazzz) [reply Makefile:5]: you need to install[ mvn](https://maven.apache.org/download.cgi) add it to the path, its similar to npm 👍🏾

File: frontend/src/utils/config.ts (hunk 2 above)
Comment (arturoLucazzz) [reply frontend/src/utils/config.ts:6]: Ive just saw this, srry jeje, THAT is a nice point of view and im in favour, but I also think that API independence is API gateway's responsibility (can be this an idea for another project? advanced api versioning 😄), I understand that it is not in the scope for this project so I can rollback it, thanks ✌🏾

**Comment by HYAR7E** [comment]
Closing this since it's so bad I don't want to even see it anymore

**Pull request closed without merging**